lib/Conversion/TritonGPUToLLVM/ReduceOpToLLVM.cpp

```diff
@@ -123,6 +141,17 @@ struct ReduceOpConversion
         // Same as BlockedEncodingAttr case
         writeIdx[axis] = udiv(index[axis], axisSizePerThread);
       }
+    } else if (auto mfmaLayout = layout.dyn_cast<MfmaEncodingAttr>()) {
+      if (axis == 0) {
+        // Because warpTileSize = [32, 32] and threadsPerWarp = [2, 32], each 2
+        // rows in smem would correspond to a warp. The mapping
+        // is: (warp_index) x 2 + (row index within warp)
+        writeIdx[axis] = add(mul(udiv(index[axis], _32), _2),
```

Comment:
It would be better to pull out the code in MfmaEncodingAttr branch so that we have a unified interface for both platforms. This will reduce the duplicate of the same code.